### PR TITLE
Enable ACME support in the docker image

### PR DESCRIPTION
**Pull request by richvdh**
Also:
 * Fix wrapping in docker readme
 * Clean up some docs on the docker image
 * a workaround for #4554

### CODE REVIEW

**Comment by richvdh** [comment]
fixes https://github.com/matrix-org/synapse/issues/4548 I hope

File: docker/README.md [hunk 1]
```diff
@@ -53,6 +52,27 @@ In order to setup an application service, simply create an ``appservices``
 directory in the data volume and write the application service Yaml
 configuration file there. Multiple application services are supported.
 
+## TLS certificates
+
+Synapse requires a valid TLS certificate. You can do one of:
```
Comment (anoadragon453) [inline docker/README.md]:
```suggestion
Synapse requires a valid TLS certificate. You can do one of the following:
```

File: docker/README.md [hunk 2]
```diff
@@ -53,6 +52,27 @@ In order to setup an application service, simply create an ``appservices``
...
+
+ * Provide your own certificate and key (as
+   `${DATA_PATH}/${SYNAPSE_SERVER_NAME}.crt|key`, or elsewhere by providing
+   an entire config as `${SYNAPSE_CONFIG_PATH`).
```
Comment (anoadragon453) [inline docker/README.md]:
```suggestion
   an entire config as `${SYNAPSE_CONFIG_PATH}`).
```

File: docker/README.md [hunk 3]
```diff
@@ -53,6 +52,27 @@ In order to setup an application service, simply create an ``appservices``
 directory in the data volume and write the application service Yaml
 configuration file there. Multiple application services are supported.
 
+## TLS certificates
+
+Synapse requires a valid TLS certificate. You can do one of:
+
+ * Provide your own certificate and key (as
+   `${DATA_PATH}/${SYNAPSE_SERVER_NAME}.crt|key`, or elsewhere by providing
```
Comment (anoadragon453) [inline docker/README.md]: Is `|key` supposed to be an actual  key? If so, then should it be `|${key}`?

File: docker/README.md [hunk 4]
```diff
@@ -53,6 +52,27 @@ In order to setup an application service, simply create an ``appservices``
...
+ * Use a reverse proxy to terminate incoming TLS, and forward the plain http
+   traffic to port 8008 in the container. In this case you should set `-e
+   SYNAPSE_NO_TLS=1`.
+
+ * Use the ACME (Let's Encrypt) support built into Synapse. This requires
+   `${SYNAPSE_SERVER_NAME}` port 80 to be forwarded to port 8009 in the
+   container, for example with `-p 80:8009`. To enable it in the docker
+   container, set `-e SYNAPSE_ACME=1`.
+
+If you don't do any of these, Synapse will fail to start with an error like:
```
Comment (anoadragon453) [inline docker/README.md]:
```suggestion
If you don't do any of these, Synapse will fail to start with an error similar to:
```

File: docker/README.md (hunk 3 above)
Comment (richvdh) [reply docker/README.md]: clarified, I hope.

Comment (anoadragon453) [reply docker/README.md]: Much, thank you :)

**Pull request merged**